indico/htdocs/js/indico/modules/timetable/timetable/Actions.js

```diff
@@ -178,112 +178,76 @@ type("TimetableManagementActions", [], {
         var self = this;
         var info = new WatchObject();
 
-        info.set('scheduleEntry', eventData.scheduleEntryId);
-        info.set('conference', eventData.conferenceId);
         info.set('startDate', startDate);
         info.set('endDate', endDate);
         info.set('reschedule', reschedule);
         info.set('sessionTimetable', this.isSessionTimetable);
 
-        var type = eventData.entryType;
-        if(type == undefined) {
-          type = "Contribution";
-        }
-
         if (exists(eventData.sessionId)) {
             info.set('session', eventData.sessionId);
             info.set('slot', eventData.sessionSlotId);
-
-            if (type != 'Session') {
-                type = 'Session' + eventData.entryType;
-            } else if (!self.isSessionTimetable){
-                type = 'SessionSlot';
-            }
         }
 
-        var dfr = $.Deferred();
-
-        //Displays a "loading ..." div at the righthand bottom side
-        var killProgress = IndicoUI.Dialogs.Util.ttStatusInfo($T('saving...'));
-
-        indicoRequest(this.methods[type].modifyStartEndDate, info, function(result, error){
-            if (!undo) {
-                killProgress();
-            }
-
-            if (error) {
-                IndicoUtil.errorReport(error);
-                dfr.reject(error);
-                self.timetable.timetableDrawer.redraw();
-                if (undo) {
-                    killProgress();
+        return $.ajax({
+            url: build_url(Indico.Urls.Timetable.entry.changeDatetime, {confId: eventData.conferenceId[0],
+                                                                        timetable_entry_id: eventData.scheduleEntryId}),
+            method: 'POST',
+            data: info.getAll(),
+            error: handleAjaxError,
+            complete: IndicoUI.Dialogs.Util.progress(),
+            success: function(data) {
+                if (!data.success) {
```

Comment:
Why can't we raise a proper error that doesn't come with a 200 status code? I added  `UserValueError` recently in case you need a NoReportError-like error that uses an error status code.